### ISSUE TITLE
DocumentStore

### ISSUE THREAD

**Issue by haadcode** [haadcode/orbit-db#56]
Add a new datastore to orbit-db: Document Store. 

Original discussion here: https://github.com/haadcode/orbit-db/issues/35

**Comment by shamb0t** [comment]
I can work on this

**Comment by haadcode** [comment]
Excellent! Let's go through today what needs to be done. Assigning the issue to you.

**Comment by haadcode** [comment]
API idea:
```javascript
const store = orbitdb.documents('db name', { indexBy: 'title' })
store.put({ title: 'hello world', doc: 'all the things'})
store.get('hello world') // return the object ^
```

**Comment by haadcode** [comment]
And we might want to extend that to:
```javascript
store.search({ doc: 'things'}) // return the document in the example above
```

This is purely optional for now.

**Comment by haadcode** [comment]
@shamb0t Any progress on this? Do you still plan to work on this? If not, let me know and I'll move the task back to the Backlog (instead of "this will be worked on soon").

**Comment by shamb0t** [comment]
@haadcode just for exact title matches so far

**Comment by haadcode** [comment]
Published to npm as version 0.13.0, https://github.com/haadcode/orbit-db/commit/625748e68bf7164a9fa54dd8fb2aa1a8d4f442a2.

<3

\o/

**Comment by haadcode** [comment]
Thank you @shamb0t for implementing this, great work and a great new feature for orbit-db! \o/

🎉 💥
